lib/dataset.js

```diff
@@ -219,16 +223,15 @@ Dataset.prototype.stream = function (options) {
       acc = '';
     }
     return acc;
-  }
+  };
 
-  function combiner(acc1, acc2) {
+  const combiner = (acc1, acc2) => {
     var p = acc2.path.match(/.+\/([0-9]+)/)[1];
     pstreams[p] = self.sc.getReadStreamSync(acc2);
-  }
+  };
 
-  this.aggregate(reducer, combiner, '', opt, function () {
```

Comment:
Please do not replace of `for (var i = 0; ....)` by `forEach`. It's outside of scope of this PR, and may have performance impact which should be analyzed.